solutions/usaco-861.mdx

```diff
@@ -0,0 +1,125 @@
+---
+id: usaco-860
+source: USACO Gold 2018 December
+title: Fine Dining
+author: Nathan Gong
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_dining_gold_dec18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(M\log N)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class FineDining {
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("dining.in"));
+		PrintWriter out = new PrintWriter("dining.out");
```

Comment:
Kattio?

Reply:
Scanner works fine in this problem, so there is no need for it